File: engine/engine-model/src/main/scala/tools/packet/alliance/UpdateAllianceInfo.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import net.server.guild.MapleAlliance
import tools.packet.PacketInput

class UpdateAllianceInfo(private var _alliance: MapleAlliance, private var _worldId: Int) extends PacketInput {
  def alliance: MapleAlliance = _alliance

  def worldId: Int = _worldId

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/UpdateAllianceInfo.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import net.server.guild.MapleAlliance;
import tools.packet.PacketInput;

public record UpdateAllianceInfo(MapleAlliance alliance, Integer worldId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
